presto-main/src/main/java/com/facebook/presto/type/TimeWithTimeZoneOperators.java

```diff
@@ -114,6 +117,18 @@ public static Slice castToSlice(@SqlType(StandardTypes.TIME_WITH_TIME_ZONE) long
         return utf8Slice(printTimeWithTimeZone(value));
     }
 
+    @ScalarOperator(CAST)
+    @SqlType(StandardTypes.TIME_WITH_TIME_ZONE)
+    public static long castFromSlice(ConnectorSession session, @SqlType(StandardTypes.VARCHAR) Slice value)
```

Comment:
I do not know about such possibility.
Implicit cast will work, as the TIME (without TZ) is most specific function for the pair (TIME, TIME_WITH_TIME_ZONE) and now it can parse string with timezone. This is the reason of the change here: https://github.com/Teradata/presto/pull/129/files#diff-61e5a60eb4040ce099bf4fe133374e6aL125

Explicit cast is explicit so there is no ambiguity AFAIU.